### PR TITLE
Category implementation, Travis CI config, sitemap generation, and photo credits added.

### PR DESCRIPTION
**Pull request by amanabt**
[Build Status](https://travis-ci.org/amanabt/startbootstrap-clean-blog-jekyll)
See demo: https://amanabt.github.io/startbootstrap-clean-blog-jekyll/
Link to the generated sitemap: https://amanabt.github.io/startbootstrap-clean-blog-jekyll/sitemap.xml
See My blog: https://amanabt.github.io/blog

**Modifications:**
- Website modified to react to the classification of a post in certain
  categories. Modifications include:
  * Using `jekyll-archives` for categories
  * Separate pages to list the posts from each `categories`
  * A post can be in multiple or no category
  * Now the `Posts` link in the navbar is a dropdown menu which
    which automatically lists the categories of the posts. Additional `css`
    for the styling of the dropdown menu also added
- Option to activate `google analytics` added.
- Option to display background photo credits added for page and home
  layouts.
- Option to generate `sitemaps.org` compliant sitemap for the `Jekyll` site
  added.
- Travis CI config and script to check the build of the website added.
  - [Build Status](https://travis-ci.org/amanabt/startbootstrap-clean-blog-jekyll)
  - You will need to config Travis-CI for your repo to get this working.
- Option to add `LinkedIn` and `Goodreads` profile links at the footer.

### CODE REVIEW

File: _config.yml [hunk 1]
```diff
@@ -3,12 +3,17 @@ email:             your-email@example.com
 description:       A Blog Theme by Start Bootstrap
 author:            Start Bootstrap
 baseurl:           "/startbootstrap-clean-blog-jekyll"
-url:               "http://blackrockdigital.github.io"
+url:               "http://amanabt.github.io"
```
Comment (coliff) [inline _config.yml]: I don't think you should change the sites URL here

Comment (amanabt) [reply _config.yml]: I changed the site URL temporarily so as to host a demo website. If you are fine with all other changes, then I will revert the `url` to `http://blackrockdigital.github.io` and put up a `READY TO MERGE` flag.

Comment (coliff) [reply _config.yml]: ah ok - sorry. I'm not a member of this repo, but just recently discovered this theme and was looking through the PRs. This looks like a good one so hopefully @davidtmiller @BlackrockDigital can review and merge. :-)

**Comment by davidtmiller** [comment]
Very good additions, thanks - sorry it's taken me a while to get to these. Thanks @amanabt for the new features and to @coliff for the review. I'll polish some of these new additions up style-wise for the demo, but this is a great set of features.

**Comment by davidtmiller** [comment]
After reviewing a few of these changes, the category implementation is a bit rough - I've remove it for now. It's a feature that everyone may not want, same with the background image. Keeping most of the rest of it though.

**Comment by amanabt** [comment]
prism.js is for colour coding the code blocks in a blog

**Comment by amanabt** [comment]
I will have a look at the category implementation and see if I can make it better. Thanks for taking time to review the code and your suggestions

**Pull request merged**